buffer/src/main/java/io/netty5/buffer/api/adaptor/ByteBufBuffer.java

```diff
@@ -613,6 +709,9 @@ public int getUnsignedByte(int roff) {
 
     @Override
     public Buffer writeByte(byte value) {
+        if (readOnly()) {
+            throw bufferIsReadOnly(this);
+        }
```

Comment:
should we better do these changes in another PR and add the relevant tests there ? This seems completely unrelated to this change

Reply:
https://github.com/netty/netty/pull/12355

Reply:
When that one is merged, I can rebase this PR on top. The component iteration tests run into this problem, but in a more obscure way, so it needs to be fixed before this PR can go in.